galley/cmd/galley/cmd/server.go

```diff
@@ -97,80 +145,128 @@ func serverCmd() *cobra.Command {
-		serverArgs.SinkMeta, "Comma-separated list of key=values to attach as metadata to outgoing sink connections. Ex: 'key=value,key2=value2'")
+	serverCmd.PersistentFlags().StringSlice("sinkMeta",
+		serverArgs.SinkMeta, "Comma-separated list of key=values to attach as metadata to outgoing sink connections. Ex'key=value,key2=value2'")
 
 	// validation config
-	serverCmd.PersistentFlags().StringVar(&validationArgs.WebhookConfigFile,
+	serverCmd.PersistentFlags().String(
 		"validation-webhook-config-file", "",
 		"File that contains k8s validatingwebhookconfiguration yaml. Validation is disabled if file is not specified")
-	serverCmd.PersistentFlags().UintVar(&validationArgs.Port, "validation-port", 443,
+	serverCmd.PersistentFlags().Uint("validation-port", 443,
 		"HTTPS port of the validation service. Must be 443 if service has more than one port ")
-	serverCmd.PersistentFlags().BoolVar(&validationArgs.EnableValidation, "enable-validation", validationArgs.EnableValidation,
+	serverCmd.PersistentFlags().Bool("enable-validation", validationArgs.EnableValidation,
 		"Run galley validation mode")
-	serverCmd.PersistentFlags().StringVar(&validationArgs.DeploymentAndServiceNamespace, "deployment-namespace", "istio-system",
+	serverCmd.PersistentFlags().String("deployment-namespace", "istio-system",
 		"Namespace of the deployment for the validation pod")
-	serverCmd.PersistentFlags().StringVar(&validationArgs.DeploymentName, "deployment-name", "istio-galley",
+	serverCmd.PersistentFlags().String("deployment-name", "istio-galley",
 		"Name of the deployment for the validation pod")
-	serverCmd.PersistentFlags().StringVar(&validationArgs.ServiceName, "service-name", "istio-galley",
+	serverCmd.PersistentFlags().String("service-name", "istio-galley",
 		"Name of the validation service running in the same namespace as the deployment")
-	serverCmd.PersistentFlags().StringVar(&validationArgs.WebhookName, "webhook-name", "istio-galley",
+	serverCmd.PersistentFlags().String("webhook-name", "istio-galley",
 		"Name of the k8s validatingwebhookconfiguration")
 
+	// Hidden, file only flags for validation specific TLS
+	serverCmd.PersistentFlags().String("validation.tls.clientCertificate", "",
+		"File containing the x509 Certificate for HTTPS validation.")
+	serverCmd.PersistentFlags().MarkHidden("validation.tls.clientCertificate")
+	serverCmd.PersistentFlags().String("validation.tls.privateKey", "",
+		"File containing the x509 private key matching --validation.tls.clientCertificate.")
+	serverCmd.PersistentFlags().MarkHidden("validation.tls.privateKey")
+	serverCmd.PersistentFlags().String("validation.tls.caCertificates", "",
+		"File containing the caBundle that signed the cert/key specified by --validation.tls.clientCertificate and --validation.tls.privateKey.")
+	serverCmd.PersistentFlags().MarkHidden("validation.tls.caCertificates")
+
 	serverArgs.IntrospectionOptions.AttachCobraFlags(serverCmd)
```

Comment:
So what about these options, and the logging options below? These end up being only available on the command-line and not in the config file, correct?

Reply:
I think we should add (file) support for  these as well.

Reply:
These options are available as-is from the config file in a flat structure.  They do not have deeply nested aliases, though we could add something like that to each config file if we want...